src/org/python/core/PyMemoryView.java

```diff
@@ -67,8 +67,30 @@ static PyObject memoryview_new(PyNewWrapper new_, boolean init, PyType subtype,
         ArgParser ap = new ArgParser("memoryview", args, keywords, "object");
         PyObject obj = ap.getPyObject(0);
 
-        if (obj instanceof BufferProtocol) {
-            return new PyMemoryView((BufferProtocol)obj);
+
+        /*
+         * Memoryview in 2.7 throws typeerror on unicode rather than assuming encoding like buffer().
+         * Various other array derivitives whos content encodeing can't be verified do the same. Reliable typing
+         * is one of the features of memoryview over the older buffer Py2kbuffer, and throwing the right exception
+         * is used in downstream modules like msgpack and pip even if technically there is a buffer interface behind
+         * the target object in our instance.
+         *
+         * This still accepts basically all of the PyArray typed variants, but probably shouldn't. Adjusting type
+         * string can granularly remove them but we should make sure there is not an active
+         * use case for this extra functionality before we get rid of it, and optionally allow a flag
+         * to ignore the type check if we really need memoryviews of array.arrays.
+        */
+        String accept_array_types = "zcbBhHiIlLduf";
+        if (obj instanceof Py2kBuffer || obj instanceof PyMemoryView || obj instanceof PyByteArray ||
```

Comment:
The idea of the buffer protocol is that you already know how to talk to objects of types you haven't met. Therefore, an attempt to enumerate the acceptable types is misguided. The aim is to accept any type implementing `BufferProtocol`, but sometimes we have to exclude types whose claim is spurious. (One day, `PyUnicode` will no longer extend `PyString`.) As mentioned, `PyArray` is to be addressed separately

Without PyArray, all we need here is
```
if (obj instanceof BufferProtocol && !(obj instanceof PyUnicode)) { ...
```